src/app/screens/Options/TestConnection/index.tsx

```diff
@@ -68,6 +50,21 @@ export default function TestConnection() {
     }
   }
 
+  /**
+   * update AccountContext, but handle currentAccountId separately,
+   * to keep TestConnection account,
+   * even though the account menu might be changed
+   */
+  const handleSelectAccount = async () => {
+    const { currentAccountId } = await api.getStatus();
+    await msg.request("selectAccount", {
+      id: currentAccountId,
+    });
+    setAccountId(currentAccountId);
+    setCurrentAccountId(currentAccountId);
```

Comment:
I don't understand this, why do we need another variable to track the current account? when we create a new account it should be selected, and we test that one. I don't think a user would change the account while the connection is being tested.

Reply:
@rolznz what should happen when the user changes the selected account in the AccountMenu? My intention was to not change the screen data, because I thought the screen refers to the just before created Account. If you think it makes more sense for the screen info to reflect whatever account is or will be selected in in the AccountMenu, we can simplify this by just jusing accound.id directly.

Reply:
Sorry, I misunderstood. I see this is for the Options version of TestConnection. I don't know what should happen in this case, probably it should navigate away from this page if the user selects a different account.

I am now thinking it might be better to have these TestConnection changes in a separate PR and keep this one as just replacing the custom SWR code with the SWR library. I think your were correct in saying that it isn't that important since this is only used once in setup.

Reply:
@rolznz so we keep as is for now?

Reply:
Maybe, or we could revert the TestConnection changes for now and put them in a separate PR later, once this one is merged.

Reply:
I really recommend not to revert back. We would need to bring back a lot of code in AccountContext.fetchAccountInfo, api.swr.getAccountInfo and getAccountsCache that I already removed. I mean I can revert, back it is quite a lot of code and for no real reason. Right now we have a lot simplified. I suggest to keep as is. What would be the reason to revert and reimplement later?

Reply:
Sounds good, let's keep it as is then.

I just suggested it to make this PR more easier to be reviewed by other Alby team members.